tests/collection.rs

```diff
@@ -1059,19 +1060,85 @@ async fn test_get_read_document() {
     let create = coll.create_document(test_doc, None).await;
     assert_eq!(create.is_ok(), true, "succeed create a document");
 
-    let _key = create.unwrap().header.unwrap()._key;
-
+    let header = create.unwrap().header.unwrap();
+    let _key = header._key;
+    let _rev = header._rev;
     let read = coll.read_document(_key.as_str()).await;
 
     let result: Document<Value> = read.unwrap();
 
     assert_eq!(result.document["no"], 1);
     assert_eq!(result.document["testDescription"], "read a document");
+    // Test if we get the right doc when it does match
+    let read: Result<Document<Value>, ClientError> = coll
+        .read_document_with_options(
+            _key.as_str(),
+            Some(DocumentReadOptions::IfMatch(_rev.clone())),
+        )
+        .await;
+    assert_eq!(read.is_err(), false);
+    // Test if we get the 412 code response when there is no match
+    let read: Result<Document<Value>, ClientError> = coll
```

Comment:
Here for some reason the error is parsed as a document, so I do not get the error.

Reply:
What is the json response? 

I think that's because this method deserialize json as T, and here `Document<Value>`. It can only throws deserialize error and cannot map to `ClientError`. Since you use `Value`, it can deserialize arbitrary json.